### PR TITLE
Slider flexbox layout

### PR DESCRIPTION
**Pull request by armenbadalyan**
Slider component is relying on` transform: [{ rotate: '90deg' }]` style to implement vertical orientation. This seems to cause issues similar to the ones described in #1281. This PR refactors the Slider component to use flexbox layout thus making Slider size and position more predictable in both vertical and horizontal orientations.

### CODE REVIEW

**Comment by iRoachie** [comment]
Can you resolve the conflicts?

**Comment by armenbadalyan** [comment]
@Monte9 , thank you for taking the time to review my PR.  In horizontal orientation, since the width in your example was not specified explicitly, according to react-native layout rules it takes 100% of the parent container width. In vertical orientation one should specify the width either explicitly or using flexbox layout. In other words it behaves just like a regular View.
Here's an example of vertically oriented slider:

```
<Slider
        orientation="vertical"
        minimumValue={0}
        maximumValue={100}
        minimumTrackTintColor="blue"
        maximumTrackTintColor="black"
        trackStyle={{ height: 7, borderRadius: 5 }}
        thumbStyle={{
          backgroundColor: "green",
          width: 30,
          height: 30,
          borderRadius: 15
        }}
        value={this.state.value}
        onValueChange={value => this.setState({ value })}
        style={{ height: 200 }}
        step={10}
      />

```

**Comment by Monte9** [comment]
Ah that makes sense. Thanks for the detailed explanation.
screen shot 2019-01-13 at 10 55 19 am

Looks good on my end. Lets just make sure we add a note on the docs mentioning that when in "vertical" orientation, height is also required.

**Comment by Monte9** [comment]
Also appreciate you submitting the PR! We use OpenCollective to give back to contributors like yourself so I'd be happy to send $20 your way for your efforts. Let me know if this sounds good to you and I can tell you how to submit your expense.

**Comment by iRoachie** [comment]
Are we waiting on the change to the docs ?

File: src/slider/Slider.js [hunk 1]
```diff
@@ -547,21 +595,37 @@ Slider.defaultProps = {
...
+    flex: 1,
+    width: TRACK_SIZE,
+  },
   thumb: {
     position: 'absolute',
     width: THUMB_SIZE,
     height: THUMB_SIZE,
     borderRadius: THUMB_SIZE / 2,
+  },
+  thumbHoizontal: {
```
Comment (markj-) [inline src/slider/Slider.js]: Small typo here. Great work with the PR. I've just stumbled upon this issue myself.

**Comment by iRoachie** [comment]
Thanks again @armenbadalyan 💯

**Comment by armenbadalyan** [comment]
Thank you guys, it was my pleasure :)

**Pull request merged**